File: Processing_M-Files/labelMaxFreqs.m
% YfreqMax = array, same length of YfreqDomain with 1 or 0 to ID peaks
% Peaks are denoted as 1.

%#eml

function YfreqMax = labelMaxFreqs (YfreqDomain)
    OverNum = 4*median(YfreqDomain); % Set cutoff for peak ID

    YfreqOver = YfreqDomain >= OverNum; % returns 1 or 0 to form over/under array
    
    %Find peak point
    YfreqMax = zeros(length(YfreqDomain),1); %record high frequencies as 1
    newPeak = 0; % flag for new peak (consecutive sequence)
    highestPoint = 0; % temp variable for highest peak point
        
    for i=2:numel(YfreqOver)-1, %@@@@@ ignoring first and last points
        if (YfreqOver(i) == 1), %In a peak
            if (newPeak == 0), %Peak is new
                newPeak = 1; %peak found
                highestPoint = (i);
            else %Peak continues
                if (YfreqDomain(i) > YfreqDomain(highestPoint)),
                    highestPoint = i;
                end
            end
        else
            if (newPeak == 1), %Peak has ended
                YfreqMax(highestPoint) = 1;
            end
            newPeak = 0; %No Longer in a peak
            highestPoint = 0;
        end
    end
end

File: Embedded_C_for_Spectrum_Digital_eZdsp_28335/labelMaxFreqs.c
/*
 * labelMaxFreqs.c
 *
 *  Created on: Apr 11, 2010
 *      Author: Jeff
 */
 #include "CableMonitoring.h"

// freqMax = array, same length of freqDomain with 1 or 0 to ID peaks

float torbenMedian();

void labelMaxFreqs(){
	float OverNum;
    int i;
    int newPeak = 0;
    int highestPoint = 0; // temp variable for highest peak point
    
	OverNum = torbenMedian();
	OverNum *= 4; // Set cutoff for peak ID

    for(i=0;i<SPECTRA_LENGTH;i++){
        if(spectraTable[i] >= OverNum){
            freqOver[i] = 1; // returns 1 or 0 to form over/under array
        }
        else{
            freqOver[i] = 0;
        }
    }
    
    //Find peak point

	//@@@@@ ignoring first and last points to ensure neighbors
    for (i=1;i<SPECTRA_LENGTH-1;i++){
		if (freqOver[i] == 1){ //In a peak
            if (newPeak == 0){ //Peak is new
                newPeak = 1; //peak found
                highestPoint = i;
            }
            else {//Peak continues
                if (spectraTable[i] > spectraTable[highestPoint]){
                    highestPoint = i;
                }
            }
		}
		else {
            if (newPeak == 1){ //Peak has ended
                freqMax[highestPoint] = 1;
            }
            newPeak = 0; //No Longer in a peak
            highestPoint = 0;
		}
    }
}
